### DISCUSSION TITLE
eslint plugin react hooks 설정에 대해

### DISCUSSION THREAD

**Discussion by WaiNaat** [woowacourse-teams/2023-pium#247] [뜨거운 토론]
저희가 eslint-plugin-react-hooks를 설치는 했는데 설정을 안했더라구요

이 플러그인을 켜면 크게 두 가지를 합니다
1. [Rules of Hooks](https://react.dev/warnings/invalid-hook-call-warning) 강제
2. [exhaustive-deps](https://react.dev/learn/synchronizing-with-effects#step-2-specify-the-effect-dependencies) 경고

이걸 켜보니까 이펙트 쪽에서 문제가 많이 나오더라구요
일단 `handleToastClose`, `useModal`, `PetPlantTimeline`, `PetRegisterForm` 여기에서 1 또는 2에 위배되는 사항들이 있는걸 확인했습니다

이슈를 하나 파서 해당 부분을 설정하고 리팩터링을 진행하는 게 어떨까요?

**Comment by bassyu** [comment]
둘다 중요한 강제나 경고로 느껴지네요! 저는 동의합니다~

**Reply by hozzijeong** [reply]
저도 추가하는거 좋다고 생각합니다. 강제 보다는 경고가 좋을 것 같긴한데, 디펜던시의 경우에 의도적으로 추가하지 않는 경우가 있을것 같은데 살짝 우려되는 부분은 경고가 있으면 build가 안되는 경우도 있지 않나요?

**Reply by WaiNaat** [reply]
> Notice that you can’t “choose” your dependencies. You will get a lint error if the dependencies you specified don’t match what React expects based on the code inside your Effect. This helps catch many bugs in your code. If you don’t want some code to re-run, [edit the Effect code itself to not “need” that dependency.](https://react.dev/learn/lifecycle-of-reactive-effects#what-to-do-when-you-dont-want-to-re-synchronize)

공식 문서에서는 "의도적으로 디펜던시를 추가하지 않는 것"은 잘못된 것이고 그럴 경우에는 디펜던시가 없도록 분리하는 것이 맞다는 입장이에요

**Reply by WaiNaat** [reply]
그리고 지금 저희는 경고가 있어도 빌드는 되는 상황인 것 같아요
저번에 메인에 '시작하기' 부분 주석처리하면서 unused import 관련 경고가 몇 개 생겼었거든요

**Reply by WaiNaat** [reply]
기본 설정상 디펜던시는 경고고 조건문 state같은 더 중요한 애들만 오류에요

**Reply by hozzijeong** [reply]
아하! 공식문서 피셜이면 저희가 좀 더 코드를 잘 짜봐야겠네요 ㅋㅋㅋ 확인했습니다!

**Comment by WaiNaat** [comment]
#250
